packages/rpc-methods/src/restricted/getBip32Entropy.ts

```diff
@@ -142,7 +182,10 @@ export function validateCaveatPaths(caveat: Caveat<string, any>) {
     });
   }
 
-  caveat.value.forEach((path) => validatePath(path));
+  caveat.value.forEach((path) => {
+    validatePath(path);
```

Comment:
Not an expert, but if the curve is `secp256k1`, and we allow fully unhardened paths, can't the snap use the extended private key to derive root key?

Reply:
Child derivation goes one way. You can't derive a parent node from a child node, regardless of how much data you have.